### ISSUE TITLE
MissingPluginException if use flutter_branch_sdk

### ISSUE THREAD

**Issue by oleksiivinogradov** [RodrigoSMarques/flutter_branch_sdk#130]
**Describe the bug**
when just add flutter_branch_sdk in app,   sharedPreferences = await SharedPreferences.getInstance();
start produce error MissingPluginException(No implementation found for method getAll on channel plugins.flutter.io/shared_preferences)

**To Reproduce**
just add to podspec   flutter_branch_sdk: ^3.5.0

**Expected behavior**
normal using

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**
android 

**Additional context**
well described in https://stackoverflow.com/questions/66944817/missingpluginexception-shared-preferences-image-picker-etc-works-when-flutt

**Comment by RodrigoSMarques** [comment]
What packages are you using in your project?

**Comment by oleksiivinogradov** [comment]
name: openbisea
description: OpenBiSea application

# The following line prevents the package from being accidentally published to
# pub.dev using `pub publish`. This is preferred for private packages.
publish_to: 'none' # Remove this line if you wish to publish to pub.dev

# The following defines the version and build number for your application.
# A version number is three numbers separated by dots, like 1.2.43
# followed by an optional build number separated by a +.
# Both the version and the builder number may be overridden in flutter
# build by specifying --build-name and --build-number, respectively.
# In Android, build-name is used as versionName while build-number used as versionCode.
# Read more about Android versioning at https://developer.android.com/studio/publish/versioning
# In iOS, build-name is used as CFBundleShortVersionString while build-number used as CFBundleVersion.
# Read more about iOS versioning at
# https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
version: 1.2.15+39

environment:
  sdk: ">=2.12.0 <3.0.0"

dependencies:
  bitcoin_flutter: ^2.0.2
  cached_network_image: ^3.1.0
  clipboard: ^0.1.3
  cupertino_icons: ^1.0.2
  equatable: ^2.0.3
  fl_chart: ^0.40.0
  fluro: ^2.0.3
  flutter:
    sdk: flutter
  flutter_localizations:
    sdk: flutter
  flutter_web_plugins:
    sdk: flutter
  intl: ^0.17.0
#  intl_translation: ^0.17.10+1
  flutter_automation: ^2.0.0
  flutter_bloc: ^7.0.1
  flutter_datetime_picker: ^1.5.1
  flutter_keychain: ^2.0.0
  flutter_lorem: ^2.0.0
  flutter_riverpod: ^0.14.0+3
  flutter_staggered_grid_view: ^0.4.0
  flutter_svg: ^0.22.0
#  flutter_web3_provider: ^1.0.6
  http: ^0.13.3
  image: ^3.0.2
  image_picker: ^0.8.1+1
  js: ^0.6.3
  json_serializable: ^5.0.2
  mime_type: ^1.0.0
  mockito: ^5.0.10
  navigation_history_observer: ^1.1.0
  objectbox: ^1.0.0
  passwd_gen: ^1.0.0
  permission_handler: ^8.1.1
  pull_to_refresh: ^2.0.0
  qr_flutter: ^4.0.0
  share_plus: ^2.1.4
  shared_preferences: ^2.0.5
  timeago: ^3.1.0
  tuple: ^2.0.0
  url_launcher: ^6.0.9
  url_strategy: ^0.2.0
  vector_math: ^2.1.0
  web3dart: ^2.1.3
  webview_flutter: ^2.0.10
  firebase_core: ^1.8.0
  firebase_analytics: ^8.3.2
  firebase_remote_config: ^0.11.0
  firebase_messaging: ^10.0.9
  firebase_dynamic_links: ^3.0.1
  material_segmented_control: ^3.1.2
  flutter_web3: ^2.1.4
  package_info_plus: ^1.3.0
  flutter_device_locale: ^0.4.0
  get: ^4.3.8
  badges: ^2.0.1
#  flutter_branch_sdk: ^3.5.0

dev_dependencies:
  build_runner: ^2.0.4
  flutter_test:
    sdk: flutter
  objectbox_generator: any
  pedantic: ^1.11.0

dependency_overrides:
  web3dart: ^2.1.3
  pointycastle: ^3.0.1
  convert: ^3.0.0
  archive: ^3.0.0
  crypto: ^2.0.3
  hex: ^0.2.0
  meta: ^1.3.0
  plugin_platform_interface: 2.0.0
#  intl: 0.17.0
#  analyzer: ^0.34.0
#  flutter_launcher_icons: ^0.9.0
analyzer:
  stron-mode: true
  language:
    enableSuperMixins: true

flutter_icons:
  image_path: "images/appstore.png"
  android: true
  ios: true
  remove_alpha_ios: true

# For information on the generic Dart part of this file, see the
# following page: https://dart.dev/tools/pub/pubspec
# The following section is specific to Flutter.
flutter:
  # The following line ensures that the Material Icons font is
  # included with your application, so that you can use the icons in
  # the material Icons class.
  uses-material-design: true
  generate: true

  # To add assets to your application, add an assets section, like this:
  # assets:
  #   - images/a_dot_burr.jpeg
  #   - images/a_dot_ham.jpeg
  assets:
    - data/binance_smart_chain_token_list.json
    - data/etherium_token_list.json
    - data/symbols.json
    - images/
    - images/coin_icons/

  # An image asset can refer to one or more resolution-specific "variants", see
  # https://flutter.dev/assets-and-images/#resolution-aware.
  # For details regarding adding assets from package dependencies, see
  # https://flutter.dev/assets-and-images/#from-packages
  # To add custom fonts to your application, add a fonts section here,
  # in this "flutter" section. Each entry in this list should have a
  # "family" key with the font family name, and a "fonts" key with a
  # list giving the asset and other descriptors for the font. For
  # example:
  # fonts:
  #   - family: Schyler
  #     fonts:
  #       - asset: fonts/Schyler-Regular.ttf
  #       - asset: fonts/Schyler-Italic.ttf
  #         style: italic
  #   - family: Trajan Pro
  #     fonts:
  #       - asset: fonts/TrajanPro.ttf
  #       - asset: fonts/TrajanPro_Bold.ttf
  #         weight: 700
  #
  fonts:
    - family: Poppins
      fonts:
        - asset: fonts/Poppins-Regular.ttf
          style: normal
        - asset: fonts/Poppins-Italic.ttf
          style: italic
        - asset: fonts/Poppins-Light.ttf
          style: normal
          weight: 200
        - asset: fonts/Poppins-LightItalic.ttf
          style: italic
          weight: 200
        - asset: fonts/Poppins-Medium.ttf
          style: normal
          weight: 500
        - asset: fonts/Poppins-MediumItalic.ttf
          style: italic
          weight: 500
        - asset: fonts/Poppins-SemiBold.ttf
          style: normal
          weight: 600
        - asset: fonts/Poppins-SemiBoldItalic.ttf
          style: italic
          weight: 600
        - asset: fonts/Poppins-Bold.ttf
          style: normal
          weight: 800
        - asset: fonts/Poppins-BoldItalic.ttf
          style: italic
          weight: 800
        - asset: fonts/Poppins-Thin.ttf
          style: normal
          weight: 100
        - asset: fonts/Poppins-ThinItalic.ttf
          style: italic
          weight: 100
  # For details regarding fonts from package dependencies,
  # see https://flutter.dev/custom-fonts/#from-packages

**Comment by oleksiivinogradov** [comment]
any updates?

**Comment by RodrigoSMarques** [comment]
> any updates?

Not.

I tried to use your pubsepec.yaml dependencies, but I was getting several version conflict errors.

I corrected one and another appeared, I didn't continue.

Your `pubspec.yaml` has gone unformatted and is bad to read.

Try pasting again, following the directions at [link](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks)


Have you made any changes to the Android project?
Do any plugins on this list (there are many) require changing from FlutterActivity to FlutterFragmentActivity?

**Comment by oleksiivinogradov** [comment]
Here is formatted. I can't remember if I do any changes in android project, but all works well until I add ur lib in podspec

```
name: openbisea
description: OpenBiSea application

# The following line prevents the package from being accidentally published to
# pub.dev using `pub publish`. This is preferred for private packages.
publish_to: 'none' # Remove this line if you wish to publish to pub.dev

# The following defines the version and build number for your application.
# A version number is three numbers separated by dots, like 1.2.43
# followed by an optional build number separated by a +.
# Both the version and the builder number may be overridden in flutter
# build by specifying --build-name and --build-number, respectively.
# In Android, build-name is used as versionName while build-number used as versionCode.
# Read more about Android versioning at https://developer.android.com/studio/publish/versioning
# In iOS, build-name is used as CFBundleShortVersionString while build-number used as CFBundleVersion.
# Read more about iOS versioning at
# https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
version: 1.2.20+44

environment:
  sdk: ">=2.12.0 <3.0.0"

dependencies:
  bitcoin_flutter: ^2.0.2
  cached_network_image: ^3.1.0
  clipboard: ^0.1.3
  cupertino_icons: ^1.0.2
  equatable: ^2.0.3
  fl_chart: ^0.40.0
  fluro: ^2.0.3
  flutter:
    sdk: flutter
  flutter_localizations:
    sdk: flutter
  flutter_web_plugins:
    sdk: flutter
  intl: ^0.17.0
#  intl_translation: ^0.17.10+1
  flutter_automation: ^2.0.0
  flutter_bloc: ^7.0.1
  flutter_datetime_picker: ^1.5.1
  flutter_keychain: ^2.0.0
  flutter_lorem: ^2.0.0
  flutter_riverpod: ^0.14.0+3
  flutter_staggered_grid_view: ^0.4.0
  flutter_svg: ^0.22.0
#  flutter_web3_provider: ^1.0.6
  http: ^0.13.3
  image: ^3.0.2
  image_picker: ^0.8.1+1
  js: ^0.6.3
  json_serializable: ^5.0.2
  mime_type: ^1.0.0
  mockito: ^5.0.10
  navigation_history_observer: ^1.1.0
  objectbox: ^1.0.0
  passwd_gen: ^1.0.0
  permission_handler: ^8.1.1
  pull_to_refresh: ^2.0.0
  qr_flutter: ^4.0.0
  share_plus: ^2.1.4
  shared_preferences: ^2.0.5
  timeago: ^3.1.0
  tuple: ^2.0.0
  url_launcher: ^6.0.9
  url_strategy: ^0.2.0
  vector_math: ^2.1.0
  web3dart: ^2.1.3
  webview_flutter: ^2.0.10
  firebase_core: ^1.8.0
  firebase_analytics: ^8.3.2
  firebase_remote_config: ^0.11.0
  firebase_messaging: ^10.0.9
  firebase_dynamic_links: ^3.0.1
  material_segmented_control: ^3.1.2
  flutter_web3: ^2.1.4
  package_info_plus: ^1.3.0
  flutter_device_locale: ^0.4.0
  get: ^4.3.8
  badges: ^2.0.1
#  flutter_branch_sdk: ^3.5.0

dev_dependencies:
  build_runner: ^2.0.4
  flutter_test:
    sdk: flutter
  objectbox_generator: any
  pedantic: ^1.11.0

dependency_overrides:
  web3dart: ^2.1.3
  pointycastle: ^3.0.1
  convert: ^3.0.0
  archive: ^3.0.0
  crypto: ^2.0.3
  hex: ^0.2.0
  meta: ^1.3.0
  plugin_platform_interface: 2.0.0
#  intl: 0.17.0
#  analyzer: ^0.34.0
#  flutter_launcher_icons: ^0.9.0
analyzer:
  stron-mode: true
  language:
    enableSuperMixins: true

flutter_icons:
  image_path: "images/appstore.png"
  android: true
  ios: true
  remove_alpha_ios: true

# For information on the generic Dart part of this file, see the
# following page: https://dart.dev/tools/pub/pubspec
# The following section is specific to Flutter.
flutter:
  # The following line ensures that the Material Icons font is
  # included with your application, so that you can use the icons in
  # the material Icons class.
  uses-material-design: true
  generate: true

  # To add assets to your application, add an assets section, like this:
  # assets:
  #   - images/a_dot_burr.jpeg
  #   - images/a_dot_ham.jpeg
  assets:
    - data/binance_smart_chain_token_list.json
    - data/etherium_token_list.json
    - data/symbols.json
    - images/
    - images/coin_icons/

  # An image asset can refer to one or more resolution-specific "variants", see
  # https://flutter.dev/assets-and-images/#resolution-aware.
  # For details regarding adding assets from package dependencies, see
  # https://flutter.dev/assets-and-images/#from-packages
  # To add custom fonts to your application, add a fonts section here,
  # in this "flutter" section. Each entry in this list should have a
  # "family" key with the font family name, and a "fonts" key with a
  # list giving the asset and other descriptors for the font. For
  # example:
  # fonts:
  #   - family: Schyler
  #     fonts:
  #       - asset: fonts/Schyler-Regular.ttf
  #       - asset: fonts/Schyler-Italic.ttf
  #         style: italic
  #   - family: Trajan Pro
  #     fonts:
  #       - asset: fonts/TrajanPro.ttf
  #       - asset: fonts/TrajanPro_Bold.ttf
  #         weight: 700
  #
  fonts:
    - family: Poppins
      fonts:
        - asset: fonts/Poppins-Regular.ttf
          style: normal
        - asset: fonts/Poppins-Italic.ttf
          style: italic
        - asset: fonts/Poppins-Light.ttf
          style: normal
          weight: 200
        - asset: fonts/Poppins-LightItalic.ttf
          style: italic
          weight: 200
        - asset: fonts/Poppins-Medium.ttf
          style: normal
          weight: 500
        - asset: fonts/Poppins-MediumItalic.ttf
          style: italic
          weight: 500
        - asset: fonts/Poppins-SemiBold.ttf
          style: normal
          weight: 600
        - asset: fonts/Poppins-SemiBoldItalic.ttf
          style: italic
          weight: 600
        - asset: fonts/Poppins-Bold.ttf
          style: normal
          weight: 800
        - asset: fonts/Poppins-BoldItalic.ttf
          style: italic
          weight: 800
        - asset: fonts/Poppins-Thin.ttf
          style: normal
          weight: 100
        - asset: fonts/Poppins-ThinItalic.ttf
          style: italic
          weight: 100
  # For details regarding fonts from package dependencies,
  # see https://flutter.dev/custom-fonts/#from-packages
```

**Comment by oleksiivinogradov** [comment]
any updates?

**Comment by RodrigoSMarques** [comment]
Your project uses a lot of plugins and Firebase. I didn't have time to configure it.

Next week I will have time to analyze.

Plugin registration issues are not directly related to the plugin.

Either it was a change to the native code, or Flutter failed to update the plugins registration code.

Please send the code for the files:

```dart
lib/generated_plugin_registrant.dart
android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java
android/app/src/main/kotlin/com/example/project_test/MainActivity.kt
```

The name of files in the Android folder may change if your code is Java or Kotlin

**Comment by oleksiivinogradov** [comment]
[generated_plugin_registrant.dart.zip](https://github.com/RodrigoSMarques/f
[GeneratedPluginRegistrant.java.zip](https://github.com/RodrigoSMarques/flutter_branch_sdk/files/7702332/GeneratedPluginRegistrant.java.zip)
lutter_branch_sdk/files/7702327/gene
[MainActivity.kt.zip](https://github.com/RodrigoSMarques/flutter_branch_sdk/files/7702335/MainActivity.kt.zip)
rated_plugin_registrant.dart.zip)

**Comment by oleksiivinogradov** [comment]
any updates?

**Comment by RodrigoSMarques** [comment]
> any updates?

I created a test project, added its pubspec.yaml dependencies

I started Firebase and called SharedPreference in main.

I was unable to reproduce the error.

I read a report in a Facebook group I join where a developer reported that he was having this error because some package was requesting changes to AndroidManifest.xml and he hadn't completed all the changes.

His project has many packages, I didn't check the necessary settings for each one.

Unfortunately I don't know what could be the reason for the error.
